File: lib/models/vehicle.dart
class Vehicle {
  final int id;
  final String make;
  final String model;
  final String name;
  final String date;
  final String location;
  final String status;
  final String battery;
  final String delay;
  final String repairStatus;
  final String availability;

  Vehicle({
    required this.id,
    required this.make,
    required this.model,
    required this.name,
    required this.date,
    required this.location,
    required this.status,
    required this.battery,
    required this.delay,
    required this.repairStatus,
    required this.availability,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['id'] ?? 0,
      make: json['make'] ?? '',
      model: json['model'] ?? '',
      name: json['name'] ?? '',
      date: json['date'] ?? '',
      location: json['location'] ?? '',
      status: json['status'] ?? '',
      battery: json['battery'] ?? '',
      delay: json['delay'] ?? '',
      repairStatus: json['repairStatus'] ?? '',
      availability: json['availability'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'make': make,
      'model': model,
      'name': name,
      'date': date,
      'location': location,
      'status': status,
      'battery': battery,
      'delay': delay,
      'repairStatus': repairStatus,
      'availability': availability,
    };
  }
}

File: src/main/java/myapp/example/demo/Entity/Vehicle.java
package myapp.example.demo.Entity;

import com.fasterxml.jackson.annotation.JsonBackReference;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "vehicle")
public class Vehicle {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String make;
    private String model;
    private String name;
    private String date;
    private String location;
    private String status;
    private String battery;
    private String delay;
    private String repairStatus;
    private String availability;

    @ManyToOne
    @JoinColumn(name = "location_id", nullable = false)
    @JsonBackReference
    private Location locationEntity;

    // Getters and setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getMake() {
        return make;
    }

    public void setMake(String make) {
        this.make = make;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getBattery() {
        return battery;
    }

    public void setBattery(String battery) {
        this.battery = battery;
    }

    public String getDelay() {
        return delay;
    }

    public void setDelay(String delay) {
        this.delay = delay;
    }

    public String getRepairStatus() {
        return repairStatus;
    }

    public void setRepairStatus(String repairStatus) {
        this.repairStatus = repairStatus;
    }

    public String getAvailability() {
        return availability;
    }

    public void setAvailability(String availability) {
        this.availability = availability;
    }

    public Location getLocationEntity() {
        return locationEntity;
    }

    public void setLocationEntity(Location locationEntity) {
        this.locationEntity = locationEntity;
    }
}